Google Cloud deployment + slight reorg. of whole Edge section (#84)
crate/cloud-getting-started: docs/edge/introduction.rst

.. _edge-disclaimer:

Introduction
============

.. NOTE::

    You must have `wget`_ and `jq`_ installed for the script to function.
    CrateDB Edge is currently in Pre-Release. CrateDB Edge and related services
    are provided on an "as is" basis and may change at any time. Crate.io

>     provides no guarantees or warrant regarding the usability or performance of

Reviewer: ```suggestion
    provides no guarantees or warranty regarding the usability or performance of
```

:ref:`cloud providers or third-party tools <edge-providers>` as described
further down in this tutorial.

.. NOTE::
    A special note about bare metal Kubernetes clusters: CrateDB Edge should
    work on any bare metal cluster, but the CrateDB instances running within
    require a load balancer for outside access. If you do not have a load
    balancer (for example `MetalLB`_), you can still access the CrateDB
    clusters within, but you will need to figure out the node ports to use.

.. _edge-signup:

Sign up
-------

To use the CrateDB Cloud software, you must first sign up. Follow the steps
outlined in :ref:`this tutorial <sign-up>` to do so.

.. _edge-create-org:

Create an organization
----------------------

When you first log in to the CrateDB Cloud Console after having created an
appropriate account, you will arrive at the Organization overview page. Here
you will be prompted to create an organization.

.. image:: ../_assets/img/cloud-create-org.png
   :alt: CrateDB Console organization creation screen

Fill out the name of the organization and click the *Create organization*
button. After a short moment, the organization will be created and you can
proceed.

> You will be taken to the Subscriptions tab of the Organization overview page.

Reviewer: The subscriptions tab has been deleted now and so this is no longer true. The user will be taken to the `dashboard`  tab after creating an organization

Configure the cluster
---------------------

Configuration
'''''''''''''

Next, go through the cluster configuration process. You will see your custom
Edge region is selected, so no region selection is necessary. You can move
directly to the cluster configuration. You can configure your desired hardware
values for CPU, RAM, and storage per node manually in the panel provided, as
well as the number of nodes you want in your cluster.

.. image:: ../_assets/img/cloud-edge-config.png
   :alt: Cluster configuration panels for CrateDB Edge

On the right the cluster scale overview shows the total hardware values for the
cluster. This is simply the number of nodes you have chosen, multiplied by the
values per node you have defined.

At the bottom of the deployment screen you can configure your account settings.
Since you have already created an organization, it does not need to be set
here. However, you can now define a project that the cluster can be deployed
in, as well as the cluster name. You also determine the database username and
password that you can use to access the cluster `Admin UI`_ later on.

.. image:: ../_assets/img/cloud-edge-settings.png
   :alt: Account settings menu

> Note that the cluster name has certain validation requirements: it may contain

Reviewer: If you wanted to add extra context here - the validation requirements are that the cluster name must be URL safe as the name is used to generate a URL on which the cluster will be available

only numbers, letters, and the dash symbol -. It must begin with a letter and
end with a letter or a number, and must be at least three characters long.

> You can also here define the backup location of your CrateDB Edge cluster. You

Reviewer: ```suggestion
You can also define the backup location of your CrateDB Edge cluster. You
```

have the option of either using the default backup location for CrateDB Cloud,
which is managed by us, or use a custom backup location that is convenient to
you. This has to be an S3 bucket or a location with an equivalent
functionality. In the latter case, you can set the access key and secret here

> as well. You can test the connection as well; keep in mind that you cannot

Reviewer: Maybe something like 
```suggestion
as well. Clicking the the Test Connection button will check whether a connection to your backup location can be established. keep in mind that you cannot
```

proceed with a custom backup location unless the connection to it is
functional.

Click *Next* at the bottom right to proceed.

Billing
'''''''

Finally, you will be taken to a new screen where you can fill out your billing
information. Our payment processing is supported by `Stripe`_. At the bottom
right you can find the cards accepted by Crate.io. When you have filled out the
necessary information, click *Deploy* below it to deploy your cluster. Do not
forget to accept financial authorization by ticking the box at the bottom.

.. image:: ../_assets/img/stripe-billing.png
   :alt: Billing information screen

> The payment and billing information you have submitted will be saved in the

Reviewer: it might be good to re-assure the customer that we don't save the card details anywhere on our infrastructure but rather request them from Stripe. So perhaps something like

```suggestion
The payment and billing information you have submitted will be available to view in the
```

Billing tab of the Organization overview screen in the CrateDB Cloud Console
(i.e., the fifth tab from the left on the same screen you arrived at).

You can at any time check your current accumulated bill at the bottom left of
the CrateDB Cloud Console screen:

.. image:: ../_assets/img/cloud-billing-meter.png
   :alt: Cloud Console billing meter

Finish up
---------

You will now be returned to the CrateDB Cloud Console, but this time to the
Cluster overview page. A popup menu will remind you of the username and
password you selected for connecting to the cluster. Make sure you copy this
information to a safe place (e.g., a password manager), as it will not be
retrievable past this point.

You can use the Cluster overview page to access your cluster via the Admin UI
(see, however, the note below).

.. NOTE::
    If your Kubernetes cluster does not provide a load balancer with an
    external IP address, you will not be able to access your cluster from the
    CrateDB Cloud Console.

.. _edge-cloud-region:

Use a cloud provider region
---------------------------

> Besides creating your own custom region, it is also possible to use CrateDB

Reviewer: ```suggestion
Aside from creating your own custom region, it is also possible to use CrateDB
```

Edge in combination with an existing cloud provider. To deploy a cluster in
this way, follow the initial steps described above until you have :ref:`created
an organization <edge-create-org>`. Then, go to the Regions tab and instead of
creating a custom region, choose a cloud provider from the fields provided and
click *Deploy cluster*. You will be referred to the subscription plan screen.
Select your desired plan and proceed to the :ref:`configuration wizard
<edge-config>` as described above.

.. _edge-delete-region:

Delete a custom region
----------------------

In order to delete a custom region, click the trashcan icon at the bottom right

> of the custom region panel. A confirmation screen will appear warning that

Reviewer: ```suggestion
of the custom region panel. A confirmation dialog will appear warning that
```
